openedx/core/djangoapps/programs/utils.py

```diff
@@ -1028,3 +1054,51 @@ def is_user_enrolled_in_program_type(user, program_type_slug, paid_modes_only=Fa
         elif course_run_id in course_runs:
             return True
     return False
+
+
+def get_subscription_api_client(user):
+    """
+    Returns an API client which can be used to make Subscriptions API requests.
+    """
+    scopes = [
+        'user_id',
+        'email',
+        'profile'
+    ]
+    jwt = create_jwt_for_user(user, scopes=scopes)
+    client = requests.Session()
+    client.auth = SuppliedJwtAuth(jwt)
+
+    return client
+
+
+def get_programs_subscription_data(user, program_uuid=None):
+    """
+    Returns the subscription data for a user's program if uuid is specified
+    else return data for user's all subscriptions.
+    """
+    client = get_subscription_api_client(user)
+    api_path = f"{settings.SUBSCRIPTIONS_API_PATH}"
+    subscription_data = []
+
+    log.info(f"B2C_SUBSCRIPTIONS: Requesting Program subscription data for user: {user}" +
+             (f" for program_uuid: {program_uuid}" if program_uuid is not None else ""))
+
+    try:
+        if program_uuid:
+            response = client.get(api_path, params={'resource_id': program_uuid, 'most_active_and_recent': True})
+            response.raise_for_status()
+            subscription_data = response.json().get('results', [])
+        else:
+            next_page = 1
+            while next_page:
+                response = client.get(api_path, params=dict(page=next_page))
+                response.raise_for_status()
+                subscription_data.extend(response.json().get('results', []))
+                next_page = response.json().get('next')
+    except Exception as exc:  # pylint: disable=broad-except
+        log.exception(
+            f"B2C_SUBSCRIPTIONS: Failed to retrieve Program Subscription Data with error: {exc}"
+            + f" for user: {user} and for program_uuid: {str(program_uuid)}" if program_uuid is not None else ""
+        )
```

Comment:
I believe we should be able to write this without string concatenation and without the ternary expression.

Reply:
Same reason as mentioned above.